test/test_io.py

```diff
@@ -0,0 +1,334 @@
+                4278190080,
+                b"\x00\x00\x00\xff",
+                b"\xff\x00\x00\x00",
+            ),
+            (
+                "qword",
+                "i64",
+                -9223372036854775808,
+                b"\x00\x00\x00\x00\x00\x00\x00\x80",
+                b"\x80\x00\x00\x00\x00\x00\x00\x00",
+            ),
+            (
+                "unsignedQword",
+                "u64",
+                18374686479671623680,
+                b"\x00\x00\x00\x00\x00\x00\x00\xff",
+                b"\xff\x00\x00\x00\x00\x00\x00\x00",
+            ),
+        ):
+            # First naming
+            check_method(name=_[0], value=_[2], binary=_[3], littleEndian=True)
+            check_method(name=_[0], value=_[2], binary=_[4], littleEndian=False)
+            # Second naming
+            check_method(name=_[1], value=_[2], binary=_[3], littleEndian=True)
+            check_method(name=_[1], value=_[2], binary=_[4], littleEndian=False)
+
+        # For float we need to check if values are almost equal
+        check_method(
+            "floating",
+            3.40282346639e38,
+            b"\xff\xff\x7f\x7f",
+            True,
+            float_equal(0.00001e38),
+        )
+        check_method(
+            "floating",
+            3.40282346639e38,
+            b"\x7f\x7f\xff\xff",
+            False,
+            float_equal(0.00001e38),
+        )
+        check_method(
+            "double",
+            1.7976931348623157e308,
+            b"\xff\xff\xff\xff\xff\xff\xef\x7f",
+            True,
+            float_equal(0.00001e38),
+        )
+        check_method(
+            "double",
+            1.7976931348623157e308,
+            b"\x7f\xef\xff\xff\xff\xff\xff\xff",
+            False,
+            float_equal(0.00001e38),
+        )
+
+        check_method("sz754", "hello", b"\x05\x00\x00\x00hello\x00", True)
+        check_method("sz754", "hello", b"\x00\x00\x00\x05hello\x00", False)
+
+        # cString methods
+        for _ in ["cString", "cStringA", "cStringU"]:
+            check_method(_, "hello", b"hello\x00", False)
+
+        # textLine methods
+        for _ in ["textLine", "textLineA", "textLineU"]:
+            check_method(_, "hello", b"hello\n", False)
+
+    def test_sz754(self):
+        """Test reading/writing sz754."""
+        io = self._create_io()
+
+        # Write hello => b'\x00\x00\x00\x05hello\x00'
+        io.sz754 = "hello"
+        self._check_len(io, 10)
+        self._check_index(io, 10)
+
+        # Write world => b'\x00\x00\x00\x05world\x00'
+        io.sz754 = "world"
+        self._check_len(io, 20)
+        self._check_index(io, 20)
+
+        # Read hello
+        io.index = 0
+        self.assertEqual(io.sz754, "hello", "incorrect value")
+        self._check_index(io, 10)
+
+        # Read world
+        self.assertEqual(io.sz754, "world", "incorrect value")
+        self._check_index(io, 20)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Do you mind me asking why you chose unittest over pytest? it's really for my own interest rather than being nitpicky 

I tend to lean towards pytest cos I'm a bit funny about oop for the sake of it but there's a good chance I'm missing something

Though could you swap the spaces out for tabs just so it's consistent with the rest of the codebase? (it's a doddle in the likes of vscode)

Thanks